Add page on Menubar and World menu (#140)
pharo-open-documentation/pharo-wiki: General/MenuBar.md

# Menubar and World menu

In Pharo, to open tools or execute actions a Menubar (At the top of the screen) and a WorldMenu (Appears on a left click in the world) are present.

> This page will document sone aspects of those features. 

Reviewer: ```suggestion
This page will document some aspects of those features. 
```

- [Disable the Menubar/WorldMenu](#disable-the-menubarworldmenu)
- [Add your own entries](#add-your-own-entries)
  * [Add a simple entry](#add-a-simple-entry)
  * [Add a sub menu](#add-a-sub-menu)
- [Change the menu](#change-the-menu)

## Disable the Menubar/WorldMenu

> In some case, we might want to hide the menubar and/or the world menu. For example, when releasing a new verison of a project to deploy, the development tools needs to be hidden. 

Reviewer: ```suggestion
In some case, we want to hide the menubar and/or the world menu. For example, when releasing a new version of a project to deploy, the development tools need to be hidden. 
```

> It is possible to do so with some settings.

Reviewer: ```suggestion
It is possible to do that with some settings.
```

**Disable Menubar**

This option is named `Show menubar` in the setting browser and is equivalent of executing:

```Smalltalk
MenubarMorph showMenubar: false
```

**Disable the WorldMenu**

It is not possible to disable this feature alone (without disabling the Menubar) in Pharo <= 7. 

In Pharo 8 it is possible to disable the WorldMenu alone with the setting named `Show world menu` or by executing:

```Smalltalk
PasteUpMorph shouldShowWorldMenu: false
```

**Disable WorldMenu and Menubar**

> To disable both the menubar and the world menu it is possible to disable both of them individually, or it is also possible to change the setting `World menu pragma` to set an empty pragma. programmatically, it can be done this way:

Reviewer: ```suggestion
To disable both the menubar and the world menu it is possible to disable each of them individually, or to change the setting `World menu pragma` to set an empty pragma. Programmatically, it can be done this way:
```

```Smalltalk
WorldState desktopMenuPragmaKeyword: ''
```

## Add your own entries

It is possible to add entries to the world menu and menu bar.

To build the menu, Pharo uses a pragma mecanism. Each class method containing a pragma `<worldMenu>` will be executed with a menu builder as parameter. The method should then configure the builder to add the needed entries.

### Add a simple entry

> To add a simple entry, you need to create and configure a new item in the builder.

Reviewer: ```suggestion
To add a simple entry, one need to create and configure a new item in the builder.
```
